### ISSUE TITLE
Test set acc doesn't work

### ISSUE THREAD

**Issue by Salidor** [fuzimaoxinan/Tensorflow-Deep-Neural-Networks#18]
I am setting the line test_Y to true:
```
# for 'classification'
        self.loss_and_acc=None     # loss, train_acc, test_acc, spend_time
        **self.test_Y=True# real label**
        self.real_class = None
        self.pred_class = None
```
But it still doesn't show the accuracy of test set

**Comment by fuzimaoxinan** [comment]
test_Y in here is not related with whether output test result or not. It just used to prevent error showing : “model has no attribution test _Y”.  It may caused by wrong setting of your dataset.

**Comment by Salidor** [comment]
What does this line is doing then?
```
if test_Y is not None:
                        acc=self.test_average_accuracy(test_X,test_Y,sess)
                        string = string + '  | 「Test」: accuracy = {:.4}%'.format(acc*100)
                        self.loss_and_acc[i][2]=acc       # <2> Test accuracy
```

**Comment by Salidor** [comment]
Also could you please provide "the code" to your dataset in encoding which all of the people around the world could read it. I've used many chinese character encodings but it didn't work. The pass is wrong.

**Comment by fuzimaoxinan** [comment]
if you provide test_Y, then prediction procedure on test set will be performed after each training epoch. By set the dataset with the form of [train_X, train_Y, test _X, test_Y], you should get the right result.

**Comment by Salidor** [comment]
![изображение](https://user-images.githubusercontent.com/25419200/61187944-758d6d80-a678-11e9-93a7-5406d4eb585a.png)
Could you please provide password to your database? As i said i've tried chinese character encodings but it didn't work.

**Comment by fuzimaoxinan** [comment]
txt already fixed, password is '7mpb'. If you don't want to process the raw sound  data with ‘wav’ form, you don't need to download it. 
in all test file, you can see the setting rule
in 'classification_MINST.py'
`datasets = [mnist.train.images,mnist.train.labels,mnist.test.images , mnist.test.labels]`
in 'classification_USC.py'
`datasets = read_data(meth='mfcc') 
#X_train, Y_train, X_test, Y_test =read_sf_data(dynamic=t)`
in 'prediction_BMS.py'
`train_X,train_Y,test_X = read_data()`

**Comment by Salidor** [comment]
Thank you for your quick answers. 

> 
> in 'classification_USC.py'
> datasets = read_data(meth='mfcc') #X_train, Y_train, X_test, Y_test =read_sf_data(dynamic=t)
When i remove comment from `X_train, Y_train, X_test, Y_test =read_sf_data(dynamic=t)` i get an error: name read_sf_data is not defined. What is sf data here?

**Comment by Tannaaaaaaaa** [comment]
I was working on this and I had the same problem of @Salidor. Can you please tell the exact procedure in order to give `Y_test` ? The classifier just goes on the training phase, never do the test because `Y_test` is always `None`. Some help please? @fuzimaoxinan

**Comment by fuzimaoxinan** [comment]
Hello, if you use the data set offered by me, it indeed will not show the results, since the data set are downloaded from the internet and it dosen't provide labels. Try to use youself data set or `classification` task. In the code, if you don't provide labels, it will not show test result in the console.

**Comment by Tannaaaaaaaa** [comment]
Thank you, I tried with some data and all worked. But could you provide me how is it possible that, with my data, I always obtain 100% test accuracy and 50-60% train accuracy? How is this possible? It seems that it is always going on overfitting.

**Comment by fuzimaoxinan** [comment]
That is confusing. Firstly, to improve train accuracy, you can appropriately adjust `learning rate`, reduce `drop out rate`, reduce `batch size`, and so on. Due to the differences in the data set, they need to try repeatedly to get the appropriate value. Then, is there data imbalance problem exiting in your data set? Is the number of your test data too small, so it is easy to get high accuracy? Finally, check your data set carefully to rule out the possibility of program errors.
